dask/bag/core.py

```diff
@@ -1616,12 +1616,20 @@ def barrier(args):
     return type(b)(merge(b.dask, dsk1, dsk2, dsk3, dsk4), name, npartitions)
 
 
+def peek(seq):
+    head = next(seq)
+    return head, cons(head, seq)
```

Comment:
Any reason to not just use http://toolz.readthedocs.io/en/latest/api.html#toolz.itertoolz.peek